serpentTools/parsers/xsplot.py

```diff
@@ -0,0 +1,109 @@
+"""Parser responsible for reading the ``*_xsplot.m`` files"""
+import numpy as np
+from serpentTools.engines import KeywordParser
+from serpentTools.messages import warning, info, debug, error
+from serpentTools.objects.readers import BaseReader
+from serpentTools.objects.xsdata import XSData
+
+def _cleanChunk(string):
+    """rstrip, but also removes = ["""
+    newstring= string.rstrip()
+    if '=' in newstring:
+        i = newstring.index('=')
+        return newstring[:i-1]
+    else:
+        return newstring
+
+class XSPlotReader(BaseReader):
+    """
+    Parser responsible for reading and working with xsplot output files.
+    These files can be generated using:
+
+    http://serpent.vtt.fi/mediawiki/index.php/Input_syntax_manual#set_xsplot
+
+    Parameters
+    ----------
+    filePath: str
+        path to the xsplot result file
+
+    Attributes
+    ----------
+    {attrs:s}
+    settings: dict
+        names and values of the settings used to control operations
+        of this reader
+    """
+
+    def __init__(self, filePath):
+        BaseReader.__init__(self, filePath, 'xsplot')
+        self.xsections = {}
+        self.metadata = {}
+
+    def _read(self):
+        """Read through the depletion file and store requested data."""
+        info('Preparing to read {}'.format(self.filePath))
+        keys = ['E', 'i[0-9]{4,5}', 'm[a-zA-Z]']
+        separators = ['\n', '];', '\r\n']
+
+        with KeywordParser(self.filePath, keys, separators) as parser:
+            for chunk in parser.yieldChunks():
+
+                if chunk[0][:5] == 'E = [':
+                    # The energy grid
+                    self.metadata['egrid']= np.array(chunk[1:], dtype=np.float64)
+
+                elif chunk[0][:15] == 'majorant_xs = [':
+                    # L-inf norm on all XS on all materials
+                    self.metadata['majorant_xs'] = np.array(chunk[1:],
+                                                             dtype=np.float64)
+
+                elif chunk[0][-7:] == 'mt = [\n':
+                    debug('found mt specification')
+                    xsname = chunk[0][:-8]
+                    isiso = True if chunk[0][0] == 'i' else False
+                    self.xsections[xsname] = XSData(xsname, self.metadata,
+                                                    isIso=isiso)
+                    self.xsections[xsname].setMTs(chunk)
+
+                elif chunk[0][-7:] == 'xs = [\n':
+                    debug('found xs specification')
+                    xsname = chunk[0][:-8]
+                    self.xsections[xsname].setData(chunk)
+
+                elif chunk[0][-7:] == 'nu = [\n':
+                    debug('found nu specification')
+                    xsname = chunk[0][:-8]
+                    self.xsections[xsname].setNuData(chunk)
+
+                else:
+                    print(chunk)
+                    error('Unidentifiable entry {}'.format(chunk[0]))
```

Comment:
Do we want the whole chunk printed here? And should this be a more severe, perhaps fatal error?

Reply:
Well, I don't think it should be an Exception or fatal error. Here's why: some strange XS may not be identified. One I came across the other day was i11023_06c_bra_f, which seems to be constant and uninteresting, somehow related to n,gamma. Maybe some sort of branching to an excited state.

There's probably some moreXS like i11023_06c_bra_f I didn't catch. So, without raising the exception, this should just skip past the chunk and continue to report only the expected stuff. As I come across these, I'll keep adding in support. Eventually I'll just sort through the source and get them all logged, but that's a bit much time for now.

Reply:
Got it got it. 

Would you be able to still store this array, perhaps in an `extras` or `others` dictionary? The reason is that while it may be uninteresting to us, there may be someone who really wants this information. We want to store all of the information present in the file. This way, if people use this reader a lot and we find out that i11023_06c_bra_f is used a lot, then we can properly treat it.

If you want, you can add some settings that would filter out the values stores. This is up to you though.